### PR TITLE
CallTileTimelineItem.CallStatus.MISSED renders both missed and reject…

### PR DESCRIPTION
**Pull request by langleyd**
Fixes https://github.com/vector-im/element-android/issues/5088

The if statement switches between `CallStatus.MISSED` and `CallStatus.ENDED`. `CallStatus.MISSED` renders both missed calls and not answered calls. So i think the logic makes more sense to check that the call was not answered vs the old code which checks all events are from one side.

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt:104]: Can I ask you to remove `!` and swap if and else blocks to simplify reading?

**Review by bmarty — APPROVED** [review]
LGTM, once again, could be nice to have @ganfra's blessing

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt:104]: was about to say the same thing :smile:

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt:104]: :). also the CI complains that the line is too long

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/CallItemFactory.kt]: Done

**Review by bmarty — APPROVED** [review]

**Pull request merged**
